subsys/mgmt/osdp/src/osdp_cp.c

```diff
@@ -241,10 +241,22 @@ static int cp_build_command(struct osdp_pd *pd, uint8_t *buf, int max_len)
 			return OSDP_CP_ERR_GENERIC;
 		}
 		assert_buf_len(CMD_KEYSET_LEN, max_len);
+		cmd = (struct osdp_cmd *)pd->ephemeral_data;
+		if (cmd->keyset.length != 16) {
+			LOG_ERR("Invalid key length");
+			return OSDP_CP_ERR_GENERIC;
+		}
 		buf[len++] = pd->cmd_id;
 		buf[len++] = 1;  /* key type (1: SCBK) */
 		buf[len++] = 16; /* key length in bytes */
-		osdp_compute_scbk(pd, buf + len);
+		if (cmd->keyset.type == 1) { /* SCBK */
```

Comment:
Consider adding and using some symbolic names (define/enum) for key types - it is better to rely on code not on comment.

Reply:
Ack. Added another patch for this.

Reply:
On second thought, I will not push it to this PR because it introduces an API change (that delays this PR from being merged). The next and final PR in this series does all the API changes, so I will queue this one there.